File: BOJ/자료구조/11723.cpp
// 11723 집합
// https://www.acmicpc.net/problem/11723
#include <iostream>

using namespace std;

int main(void) {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    int M, BIT = 0;
    string input;

    cin >> M;

    while (M) {
        int x;
        cin >> input;

        if (input == "add") {
            cin >> x;
            BIT |= (1 << x);
        } else if (input == "remove") {
            cin >> x;
            BIT &= ~(1 << x);
        } else if (input == "check") {
            cin >> x;
            if (BIT & (1 << x)) {
                cout << "1\n";
            } else cout << "0\n";
        } else if (input == "toggle") {
            cin >> x;
            BIT ^= (1 << x);
        } else if (input == "all") {
            BIT = (1 << 21) - 1;
        } else if (input == "empty") {
            BIT = 0;
        }

        M--;
    }
    
    return 0;
}

File: BOJ/자료구조/Set/11723.py
# 11723 집합
# https://www.acmicpc.net/problem/11723

import sys
def input(): return sys.stdin.readline().strip()


S = set()
M = int(input())

for _ in range(M):
    cmd = input().split()

    if len(cmd) == 1:
        if cmd[0] == "all":
            S = set([i for i in range(1, 21)])
        else:
            S = set()

        continue

    if cmd[0] == "add":
        S.add(int(cmd[1]))

    elif cmd[0] == "remove":
        S.discard(int(cmd[1]))

    elif cmd[0] == "check":
        print(1 if int(cmd[1]) in S else 0)

    elif cmd[0] == "toggle":
        if int(cmd[1]) in S:
            S.discard(int(cmd[1]))
        else:
            S.add(int(cmd[1]))
